### PR TITLE
interfaces: T3114: Fix VLAN-aware bridge setting failure

### PR DESCRIPTION
**Pull request by jack9603301**
## Change Summary

Fix VLAN-aware bridge setting failure

## Types of changes

 
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

https://phabricator.vyos.net/T3114

## Component(s) name

bridge

## Proposed changes

Due to dependency issues, if a non-existent interface appears when creating a VLAN-aware bridge, the VLAN-aware parameters of this interface cannot be set. Therefore, the interface will complete the setting when waiting for the interface to be created.

## How to test

```
set interfaces bonding bond1 member interface eth0
set interfaces bonding bond1 member interface eth1
set interfaces bridge br1 member interface bond1 native-vlan 3
set interfaces bridge br1 member interface bond1 allowed-vlan 2
commit
run show int br vlan
```

## Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

**Comment by c-po** [comment]
Randomly changing the priority is a very, very bad idea. Usually the problem lies in the configuration itself and some dependencies. The dependencies must be resolved instead and changing those parameters must probably be done here: 

https://github.com/vyos/vyos-1x/blob/current/python/vyos/ifconfig/interface.py#L1178-L1181

This is called for every interface which might be part of the bridge. So the `is_bridge_member` key holds the entire interface configuration of the member interface.

**Comment by jack9603301** [comment]
@c-po Sorry, I have changed the repair implementation

**Comment by jack9603301** [comment]
The fix is ready to merge

File: python/vyos/ifconfig/interface.py [hunk 1]
```diff
@@ -898,6 +899,82 @@ def add_to_bridge(self, bridge_dict):
...
+                    self._cmd(cmd)
+                    vlan_add.add(vlan)
+            
+            vlan_bridge_ids = Section.klass(bridge)(bridge, create=True).get_vlan_ids()
+            
+            # Delete VLAN ID for the bridge
+            for vlan in vlan_del:
+                if isinstance(vlan,str) and vlan.isnumeric():
+                    if int(vlan) == 1 and int(vlan) not in vlan_bridge_ids:
+                        cmd = f'bridge vlan del dev {bridge} vid {vlan} self'
```
Comment (c-po) [inline python/vyos/ifconfig/interface.py]: Why is this line required four times? This is too redundant. It must be shrinked to one occurance.

image

**Review by c-po — CHANGES_REQUESTED** [review]

**Comment by c-po** [comment]
Please also provide a smoketest validating this change. It must fail on the current ISO image and not fail within this PR.

File: smoketest/scripts/cli/test_interfaces_bridge.py [hunk 2]
```diff
@@ -45,17 +45,33 @@ def setUp(self):
             for tmp in Section.interfaces("ethernet"):
                 if not '.' in tmp:
                     self._members.append(tmp)
+            
+            self.session.set(['interfaces','dummy','dum0'])
+            self.session.set(['interfaces','dummy','dum1'])
+            self.session.commit()
```
Comment (c-po) [inline smoketest/scripts/cli/test_interfaces_bridge.py]: Why is this `commit()` required?

File: smoketest/scripts/cli/test_interfaces_bridge.py [hunk 3]
```diff
@@ -45,17 +45,33 @@ def setUp(self):
...
+            self.session.set(['interfaces','bonding','bond1','member','interface','dum0'])
+            self.session.set(['interfaces','bonding','bond1','member','interface','dum1'])
+            self.session.commit()
```
Comment (c-po) [inline smoketest/scripts/cli/test_interfaces_bridge.py]: Why is this `commit()` required?

File: smoketest/scripts/cli/test_interfaces_bridge.py [hunk 4]
```diff
@@ -45,17 +45,33 @@ def setUp(self):
...
+        self.session.delete(['interfaces','dummy'])
+        super().tearDown()
 
     def test_add_remove_member(self):
         """ Add member interfaces to bridge and set STP cost/priority """
         for interface in self._interfaces:
             base = self._base_path + [interface]
             self.session.set(base + ['stp'])
             self.session.set(base + ['address', '192.0.2.1/24'])
+            self.session.commit()
```
Comment (c-po) [inline smoketest/scripts/cli/test_interfaces_bridge.py]: Why is this `commit()` required?

**Review by c-po — CHANGES_REQUESTED** [review]

File: smoketest/scripts/cli/test_interfaces_bridge.py (hunk 3 above)
Comment (jack9603301) [reply smoketest/scripts/cli/test_interfaces_bridge.py]: Mainly to make the required interface effective during testing

File: smoketest/scripts/cli/test_interfaces_bridge.py (hunk 2 above)
Comment (jack9603301) [reply smoketest/scripts/cli/test_interfaces_bridge.py]: Mainly to make the required interface effective during testing

File: smoketest/scripts/cli/test_interfaces_bridge.py (hunk 4 above)
Comment (jack9603301) [reply smoketest/scripts/cli/test_interfaces_bridge.py]: I remember that I have not modified irrelevant functions, maybe I added them inadvertently

File: smoketest/scripts/cli/test_interfaces_bridge.py (hunk 3 above)
Comment (jack9603301) [reply smoketest/scripts/cli/test_interfaces_bridge.py]: This `commit` must exist, otherwise the following code will not take effect:

```
            for tmp in Section.interfaces("bonding"):
                if not '.' in tmp:
                    self._members.append(tmp)
```

Comment (c-po) [reply smoketest/scripts/cli/test_interfaces_bridge.py]: `setUp()` is called for every testcase and within every testcase we call `commit()`, so calling `commit` here is redundant, please remove.

See https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUp for more info.

**Review by c-po — CHANGES_REQUESTED** [review]

File: python/vyos/ifconfig/interface.py [hunk 5]
```diff
@@ -898,6 +899,60 @@ def add_to_bridge(self, bridge_dict):
...
+                if int(vlan_id) != 1:
+                    vlan_del.add(1)
+                cmd = f'bridge vlan add dev {self.ifname} vid {vlan_id} pvid untagged master'
+                self._cmd(cmd)
+                vlan_add.add(vlan_id)
+
+            if 'allowed_vlan' in bridge_config:
+                vlan_filter = 1
+
+            if vlan_filter:
```
Comment (c-po) [inline python/vyos/ifconfig/interface.py]: Why is this `if vlan_filter` required? Just executing the commands below under `if 'allowed_vlan' in bridge_config:` have the same effect.

**Review by c-po — CHANGES_REQUESTED** [review]
